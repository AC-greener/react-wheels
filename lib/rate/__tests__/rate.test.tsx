import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Rate from "../rate";

describe("Rate", () => {
  it("renders correctly", () => {
    const { container } = render(<Rate />);
    expect(container).toMatchSnapshot();
  });

  it("renders the correct number of stars", () => {
    const { getAllByRole } = render(<Rate count={5} />);
    const stars = getAllByRole("button");
    expect(stars.length).toBe(5);
  });

  it("fills in the correct number of stars when clicked", () => {
    const onChange = jest.fn();
    const { getAllByRole } = render(<Rate count={5} onChange={onChange} />);
    const stars = getAllByRole("button");
    fireEvent.click(stars[2]);
    expect(onChange).toHaveBeenCalledWith(3);
  });

  it("fills in the correct number of stars when hovered", () => {
    const onChange = jest.fn();
    const { getAllByRole } = render(<Rate count={5} onChange={onChange} />);
    const stars = getAllByRole("button");
    fireEvent.mouseEnter(stars[2]);
    expect(stars[0]).toHaveStyle("color: gold");
    expect(stars[1]).toHaveStyle("color: gold");
    expect(stars[2]).toHaveStyle("color: gold");
    expect(stars[3]).toHaveStyle("color: gray");
    expect(stars[4]).toHaveStyle("color: gray");
  });
});
