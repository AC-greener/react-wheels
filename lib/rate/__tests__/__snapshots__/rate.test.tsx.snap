// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rate renders correctly 1`] = `
<div>
  <div>
    <span
      style="cursor: pointer; color: gray; font-size: 24px; transition: color 0.3s;"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17.27l-6.18 3.76 1.64-7.16L2.56 9.74l7.22-.62L12 2l2.22 7.12 7.22.62-5.9 4.13 1.64 7.16z"
        />
      </svg>
    </span>
    <span
      style="cursor: pointer; color: gray; font-size: 24px; transition: color 0.3s;"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17.27l-6.18 3.76 1.64-7.16L2.56 9.74l7.22-.62L12 2l2.22 7.12 7.22.62-5.9 4.13 1.64 7.16z"
        />
      </svg>
    </span>
    <span
      style="cursor: pointer; color: gray; font-size: 24px; transition: color 0.3s;"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17.27l-6.18 3.76 1.64-7.16L2.56 9.74l7.22-.62L12 2l2.22 7.12 7.22.62-5.9 4.13 1.64 7.16z"
        />
      </svg>
    </span>
    <span
      style="cursor: pointer; color: gray; font-size: 24px; transition: color 0.3s;"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17.27l-6.18 3.76 1.64-7.16L2.56 9.74l7.22-.62L12 2l2.22 7.12 7.22.62-5.9 4.13 1.64 7.16z"
        />
      </svg>
    </span>
    <span
      style="cursor: pointer; color: gray; font-size: 24px; transition: color 0.3s;"
    >
      <svg
        fill="currentColor"
        height="1em"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 17.27l-6.18 3.76 1.64-7.16L2.56 9.74l7.22-.62L12 2l2.22 7.12 7.22.62-5.9 4.13 1.64 7.16z"
        />
      </svg>
    </span>
  </div>
</div>
`;
